clai/clai/chat/cli.py

```diff
@@ -0,0 +1,166 @@
+
+        return models, builtin_tools, builtin_tool_defs
+
+    except Exception as e:
+        print(f'Warning: Error loading config from {config_path}: {e}')
+        return None, None, None
+
+
+def select_agent(agents: list[AgentInfo]) -> AgentInfo | None:
+    """Prompt user to select an agent from the list."""
+    if not agents:
+        print('No agents found in the current directory.')
+        return None
+
+    if len(agents) == 1:
+        print(f'Found agent: {agents[0].agent_name} in {agents[0].file_path}')
+        return agents[0]
+
+    print('Multiple agents found:')
+    for i, agent_info in enumerate(agents, 1):
+        print(f'  {i}. {agent_info.agent_name} ({agent_info.file_path})')
+
+    while True:
+        try:
+            choice = input('\nSelect an agent (enter number): ').strip()
+            index = int(choice) - 1
+            if 0 <= index < len(agents):
+                return agents[index]
+            else:
+                print(f'Please enter a number between 1 and {len(agents)}')
+        except (ValueError, KeyboardInterrupt):
+            print('\nSelection cancelled.')
+            return None
+
+
+def load_agent(agent_info: AgentInfo) -> Agent | None:
+    """Load an agent from the given agent info."""
+    sys.path.insert(0, str(Path.cwd()))
+
+    try:
+        spec = importlib.util.spec_from_file_location(agent_info.module_path, agent_info.file_path)
+        if spec is None or spec.loader is None:
+            print(f'Error: Could not load module from {agent_info.file_path}')
+            return None
+
+        module = importlib.util.module_from_spec(spec)
+        sys.modules[agent_info.module_path] = module
+        spec.loader.exec_module(module)
+
+        agent = getattr(module, agent_info.agent_name, None)
+        if agent is None:
+            print(f'Error: Agent {agent_info.agent_name} not found in module')
+            return None
+
+        if not isinstance(agent, Agent):
+            print(f'Error: {agent_info.agent_name} is not an Agent instance')
+            return None
+
+        return agent  # pyright: ignore[reportUnknownVariableType]
+
+    except Exception as e:
+        print(f'Error loading agent: {e}')
+        return None
+
+
+def run_chat_command(
+    root_dir: Path | None = None,
+    host: str = '127.0.0.1',
+    port: int = 8000,
+    config_path: Path | None = None,
+    auto_config: bool = True,
+) -> int:
+    """Run the chat command to discover and serve an agent.
+
+    Args:
+        root_dir: Directory to search for agents (defaults to current directory)
+        host: Host to bind the server to
+        port: Port to bind the server to
+        config_path: Path to agent_options.py config file
+        auto_config: Auto-discover agent_options.py in current directory
+    """
+    search_dir = root_dir or Path.cwd()
+
+    print(f'Searching for agents in {search_dir}...')
+    agents = find_agents(search_dir)
```

Comment:
currently the server starts with a static agent but in the future if the UI allows it the web app could allow for swapping agents dynamically